text/204-golang-bindings.md

```diff
@@ -0,0 +1,762 @@
+}
+
+func Values() []LaunchType {
+    return []LaunchType{
+        LaunchTypeEc2,
+        LaunchTypeFargate,
+    }
+}
+```
+
+Encapsulating the enum within a package would then create an intuitive namespacing (Run example in the [Go
+playground](https://play.golang.org/p/olztr74OKsk))
+
+```go
+fmt.Println(launchtype.EC2()) // => "EC2"
+fmt.Println(launchtype.Fargate()) // => "FARGATE"
+fmt.Println(launchtype.Values()) // => [EC2 FARGATE]
+```
+
+## Interfaces
+
+### Go Interfaces
+
+Go Interfaces are named collections of *method signatures*. They can be implemented by other types, including other interfaces. Go interfaces are
+implemented *implicitly*, so, e.g., the implements keyword is not needed by the type that is implementing an interface. Interfaces can embed other
+interfaces, which effectively allows the embedding interface to “inherit” all the methods defined in the embedded interface. This is the closest to
+how typescript interfaces use extends.
+
+Interfaces can be used as fields on structs as way to encapsulate behavior and facilitate testing.
+
+### Typescript Interfaces
+
+Typescript Interfaces are types (similar to type aliases) which have *methods* and *properties*. They can *extend* other interfaces or other
+types/classes (whereas a type alias would use a type intersection instead of extends). There are three major differences between type aliases and
+interfaces:
+
+1. _Shape rather than type definition_: Define a shape rather than a collection of possible types
+2. _Extension type checking_: Extending an interface with any of the properties/methods overridden by the interface that is extending it will cause a
+   type error, whereas a type alias will do its best to create an overloaded signature. This is useful when modeling inheritance.
+3. _Declaration merging_: Multiple interfaces in the same scope with the same name are merged; multiple type aliases of the same name will throw a
+   compile-time error.
+
+Classes that implement an interface do so *explicitly* by using the implements keyword.
+
+In jsii, there are two types of interfaces; *structs* (aka: *datatypes* - immutable pure data objects) and *behavioral interfaces* (named `IXxx`,
+where `Xxx` is a resource name). The following describes how to handle each kind.
+
+### JSII Non-Datatype Interfaces (`IResource`)
+
+Typescript interfaces that define method signatures can be translated directly as Go interfaces. Any properties that the interface may contain would
+be converted to getters (and setters, as needed).
+
+**Example**:
+
+Typescript interface:
+
+```ts
+export interface ISecurityGroup extends IResource, IPeer {
+  readonly securityGroupId: string;
+  readonly allowAllOutbound: boolean;
+
+  addIngressRule(peer: IPeer, connection: Port, description?: string, remoteRule?: boolean): void;
+  addEgressRule(peer: IPeer, connection: Port, description?: string, remoteRule?: boolean): void;
+}
+```
+
+Go interface:
+
+```go
+type ISecurityGroup interface {
```

Comment:
I think the `I` prefix is convention in typescript, but it'll look out of place in Go. Maybe using some sort of `er` suffix - like SecurityGrouper? That sounds a bit odd, too.

Reply:
Yeah, it's not great but it makes it easier to match up with the source TS code for now. We can tweak this if folks complain during UX testing!